src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java

```diff
@@ -28,19 +24,20 @@
 import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
 import org.apache.sshd.server.session.ServerSession;
 import org.apache.sshd.server.shell.ShellFactory;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
 
 public class SSHAgentBase {
 
-    public static AtomicInteger SSH_SERVER_INITIAL_PORT = new AtomicInteger(4380);
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
```

Comment:
the same file was being used, which if using mutliple tests in parallel could lead to some funky timing issues also.

Reply:
Will JUnit actually apply this rule since it isn't a test suite?

Reply:
its a base class of the other tests so it should.  the test pass :)

Reply:
Meaning will the temp files created with it actually be cleaned up?

Reply:
yup!
Note that it was not cleaned up prior to this - one would always dangle.